graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -0,0 +1,1560 @@
+            The set of node type names that appear in the edge configs but
+            are missing from the node configs.
+        """
+        node_types = {node_config.ntype for node_config in node_configs}
+        missing_node_types = set()
+        for edge_config in edge_configs:
+            if edge_config.src_ntype not in node_types:
+                missing_node_types.add(edge_config.src_ntype)
+            if edge_config.dst_ntype not in node_types:
+                missing_node_types.add(edge_config.dst_ntype)
+
+        return missing_node_types
+
+    def create_node_id_maps_from_edges(
+        self, edge_configs: Sequence[EdgeConfig], missing_node_types: Optional[Set[str]] = None
+    ) -> None:
+        """Creates node id mappings from edges.
+
+        In the case where at least one node type files does exist,
+        we need to extract node ids and create node mappings from the edge files.
+
+        Given a list of edge configuration objects, this function will extract
+        all the referenced node string ids for each node type and persist them
+        so they can be used downstream during edge structure
+        creation. The secondary argument missing_node_types lists node
+        types that do not have their own node files, so
+        we need to create node id mapping for them from the edge files.
+
+        The function modifies the state of the DistHeterogeneousGraphLoader object by
+        populating the node_mapping_paths member field.
+
+        Parameters
+        ----------
+        edge_configs : Sequence[EdgeConfig]
+            A list of EdgeConfig objects that contain all edge types in the graph.
+        missing_node_types : Optional[Set[str]], optional
+            An optional set of node types that do not have corresponding node files,
+            by default None. We create mappings from the edges for the missing node
+            types.
+        """
+        # TODO: If it is possible to have multiple edge files for a single node type,
+        # we should use self.node_mapping_paths here.
+        if missing_node_types is None:
+            missing_node_types = set()
+        node_type_to_mapping_df = {}  # type: Dict[str, DataFrame]
+        for edge_config in edge_configs:
+            src_type = edge_config.src_ntype
+            src_col = edge_config.src_col
+            dst_type = edge_config.dst_ntype
+            dst_col = edge_config.dst_col
+            rel_type = edge_config.rel_type
+
+            # TODO: Read only column ids when we have that
+            edges_df = self._read_edge_df(edge_config)
+            edges_df.cache()
+
+            for ntype, ncol in [(src_type, src_col), (dst_type, dst_col)]:
+                # We only create mapping for node types that don't have a corresponding node file
+                if ntype in missing_node_types:
+                    if ntype in node_type_to_mapping_df:
+                        mapping_df = node_type_to_mapping_df[ntype]
```

Comment:
> multiple edge types may have the same src_ntype or dst_ntype?

Yes, because a single node type can appear in multiple edge files, we need to ensure we gather all the unique node ids from across all edge types to have the final complete mapping for the node type, _before_ we do the str-to-int-id conversion for the edges.